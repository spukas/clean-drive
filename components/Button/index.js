import styled from 'styled-components';

export const Button = styled.button`
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    border-radius: 3px;

    /* Color the border and text with theme.main */
    color: ${({ theme, color = 'primary' }) => theme.colors[color]};
    border: 2px solid ${({ theme, color = 'primary' }) => theme.colors[color]};
`;
