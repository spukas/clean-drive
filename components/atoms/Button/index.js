import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const gradient = color => css`
  background-image: radial-gradient(
    circle,
    ${color} 1%,
    ${color} 15%,
    rgba(0, 0, 0, 0) 30%
  );
  background-color: ${color};
`;

const floatStyles = css`
  border-radius: 50%;
  width: 50px;
  height: 50px;
  ${gradient(p => p.theme.palette.primary[0])}
  will-change: box-shadow, background-color;
  font-size: 22px;
  padding: 0;
  color: ${p => p.theme.palette.white[0]}
  &:hover {
    box-shadow: 0 5px 11px 0 rgba(0, 0, 0, 0.16), 0 4px 15px 0 rgba(0, 0, 0, 0.13);
    background-color: ${p => p.theme.palette.primary[2]};
  }
`;

const raisedStyles = css`
  border-radius: calc(3px * 0.66);
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.14), 0 2px 10px 0 rgba(0, 0, 0, 0.1);
  ${gradient(p => p.theme.palette.primary[0])};
  will-change: box-shadow, background-color;
  color: white;
  &:hover {
    box-shadow: 0 5px 11px 0 rgba(0, 0, 0, 0.16), 0 4px 15px 0 rgba(0, 0, 0, 0.13);
    background-color: ${p => p.theme.palette.primary[2]};
  }
`;

const flatStyles = css`
  ${gradient(p => p.theme.palette.primary[0])}
  background-color: white;
  box-shadow: none;
`;

const styles = css`
  transition: all 0.3s cubic-bezier(0.64, 0.09, 0.08, 1);
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.14), 0 2px 10px 0 rgba(0, 0, 0, 0.1);
  position: relative;
  cursor: pointer;
  text-transform: uppercase;
  background-image: none;
  background-size: 0;
  background-repeat: no-repeat;
  background-position: 50% 50%;
  transition: background-color 0.3s cubic-bezier(0.64, 0.09, 0.08, 1),
    box-shadow 0.3s cubic-bezier(0.64, 0.09, 0.08, 1);
  will-change: background-size, background-image;
  padding: 10px 20px;
  display: inline-block;
  font-family: ${p => p.theme.fonts.primary};
  border: 0;
  &:after {
    position: absolute;
    content: '';
    transition: none;
    background: radial-gradient(circle, white 95%, rgba(0, 0, 0, 0) 95%);
    background-size: 0.7%;
    background-position: 50% 50%;
    background-repeat: no-repeat;
  };
  &:focus {
    outline: none;
    background-size: 1000%;
    transition: all 1s cubic-bezier(0.64, 0.09, 0.08, 1);
  };

  ${props => props.float && floatStyles};
  ${props => props.raised && raisedStyles};
  ${props => props.flat && flatStyles};
`;

const StyledButton = styled.button`
  ${styles};
`;

const Button = ({ type, ...props }) => <StyledButton {...props} type={type} />;

Button.propTypes = {
  type: PropTypes.string,
};

Button.defaultProps = {
  type: 'button',
};

export { Button };
