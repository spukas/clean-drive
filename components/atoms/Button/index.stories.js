import React from 'react';
import { storiesOf } from '@storybook/react';
import { Button } from '.';

storiesOf('Button', module)
  .add('default flat', () => <Button flat>Hello</Button>)
  .add('raised', () => <Button raised>Hello</Button>)
  .add('float', () => <Button float>+</Button>);
