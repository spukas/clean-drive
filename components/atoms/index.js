export { Atom } from './Atom';
export { Block } from './Block';
export { Button } from './Button';
