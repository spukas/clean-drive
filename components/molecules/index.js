export { Navigation } from './Navigation';
export { Feature } from './Feature';