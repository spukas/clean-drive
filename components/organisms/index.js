export { Header } from './Header';
export { Hero } from './Hero';
export { Footer } from './Footer';
