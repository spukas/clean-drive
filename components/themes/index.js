import {
    breakpoints,
    space,
    fonts,
    fontWeights,
    colors,
} from './constants';

export default {
    breakpoints,
    space,
    fonts,
    fontWeights,
    colors,
};
