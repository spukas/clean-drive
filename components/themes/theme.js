import {
    breakpoints,
    space,
    fonts,
    fontWeights,
    colors,
} from './constants';

export const theme = {
    breakpoints,
    space,
    fonts,
    fontWeights,
    colors,
};
