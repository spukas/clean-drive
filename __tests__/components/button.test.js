import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Button } from '../../components/Button';
import { theme } from '../../src/theme';

describe('Button', () => {
    it('Button renders', () => {
        const component = renderer.create(<Button theme={theme} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('Button renders with secondary color prop', () => {
        const component = renderer.create(<Button theme={theme} color="secondary" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
