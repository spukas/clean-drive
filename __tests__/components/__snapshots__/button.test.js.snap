// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button renders 1`] = `
.c0 {
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
  color: mediumseagreen;
  border: 2px solid mediumseagreen;
}

<button
  className="c0"
/>
`;

exports[`Button Button renders with secondary color 1`] = `
.c0 {
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
  color: palevioletred;
  border: 2px solid palevioletred;
}

<button
  className="c0"
  color="secondary"
/>
`;

exports[`Button Button renders with secondary color prop 1`] = `
.c0 {
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
  color: palevioletred;
  border: 2px solid palevioletred;
}

<button
  className="c0"
  color="secondary"
/>
`;

exports[`Button With Snapshot Testing Button renders 1`] = `
.c0 {
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
  color: mediumseagreen;
  border: 2px solid mediumseagreen;
}

<button
  className="c0"
/>
`;
